nubia/internal/completion.py

```diff
@@ -127,6 +127,28 @@ def get_completions(self) -> Iterable[Completion]:
         Returns a
         """
         logger = logging.getLogger(f"{type(self).__name__}.get_completions")
+
+        # This is a rather sad piece of code. Since nubia uses <keyword>= as
+        # the completion string, when we specify choices for a keyword, and
+        # select the keyword from the autocompletion suggestion by hitting
+        # SPACE, the additional space confuses the autocompleter for the
+        # values associated with the space. For example, if you had state=
+        # as a keyword, and "up" and "down" as possible choices for the keyword
+        # once you've selected state= from the completion list by hitting SPACE
+        # up and down pop up as possible choices, but typing the first letter
+        # of a choice such as u(for up), doesn't trim the selection to those
+        # starting with u. Selection just stops because the parser gets confsed
+        # about the space after "keyword= " and doesn't provide any more auto
+        # complete suggestions. The lines that follow fix this by removing the
+        # space after = from the text, allowing the parser to do its job.
+        if self.doc.char_before_cursor != " ":
+            pos = self.doc.find_backwards('= ')
```

Comment:
Why look for  '= ' + char and not only have '=' + char or even just the equal sign? The space after the equal sign is not really intuitive to put.

Reply:
Its not that I'm asking for "= ". When you select a keyword from the dropdown, nubia adds the extra space after the equal to (you use SPACE to select a dropdown option). That enables only the first level of completion. For example, go to the CLI now without this patch and try the following: interface show state= where you select state= from the dropdown menu, and then select one of the options, by typing say 'n'. Instead of displaying a new list of options which only start with n, you get nothing. Thats because of the extra space after =.